### PR TITLE
Update FUTBIN Club Worth - Uses background processing and does every …

### PR DESCRIPTION
**Pull request by Kyderman**
…player in the club

Displays total club worth after about 10 seconds

### CODE REVIEW

**Comment by Mardaneus86** [comment]
Quick remark, try to adhere to the editorconfig settings. Most code editors support this. Please update this PR accordingly.

Will check functionality now.

File: futbin-club-worth.user.js [hunk 1]
```diff
@@ -13,51 +13,61 @@
 // @supportURL  https://github.com/Mardaneus86/futwebapp-tampermonkey/issues
 // ==/UserScript==
 // ==OpenUserJS==
-// @author Mardaneus86
+// @author Mardaneus86, Kyderman
```
Comment (Mardaneus86) [inline futbin-club-worth.user.js]: This is used by OpenUserJS for collaboration on their platform, no need to add your name here.

File: futbin-club-worth.user.js [hunk 2]
```diff
@@ -13,51 +13,61 @@
...
+                };
+                currentCount += 90;
+                o.addListener(communication.BaseDelegate.SUCCESS, this, function marketSearch(sender, response) {
+                    var playerIds = Object.keys(response.itemData).map(function(key) { if(!response.itemData[key].untradeable){return response.itemData[key].resourceId;}});
+                    if(response.itemData.length < 1) {
+                        clearInterval(gettingPlayerData);
+                        console.log("done");
+                         $('.clubValue .btn-subtext').html(`${currentTotal}`);
+                    }
+                    var futbinUrl = "https://www.futbin.com/18/playerPrices?player=&all_versions=" + playerIds.join(',');
```
Comment (Mardaneus86) [inline futbin-club-worth.user.js]: No need to execute if `playerIds` is an empty array.

File: futbin-club-worth.user.js [hunk 3]
```diff
@@ -13,51 +13,61 @@
...
-            </div>`);
-          }
-        });
-      }, 1000);
-    }
-  });
+              <span class="btn-subtext">Calculating, please wait...</span>
+              </div>`);
+            }, 1000);
+            var gettingPlayerData = setInterval(function() {
```
Comment (Mardaneus86) [inline futbin-club-worth.user.js]: We don't actually need the interval if we are requesting items through the delegate. Use recursion inside the SUCCESS listener in order to page. That way we can go to the next page after we processed everything from the previous iteration. It prints the 'done' message before actually calculating in my environment, so `currentTotal` is still 0 at that point.

File: futbin-club-worth.user.js [hunk 4]
```diff
@@ -13,51 +13,61 @@
...
+                var t = new transferobjects.SearchCriteria();
+                t.type = enums.SearchType.PLAYER;
+                var o = new communication.ClubSearchDelegate(t, currentCount, 90);
+                components.ClickShield.prototype.showShield = function(t) {
```
Comment (Mardaneus86) [inline futbin-club-worth.user.js]: disable-clickshield script does this already. This call would effectively disable the clickshield for the complete app.

**Comment by Kyderman** [comment]
Changes have been made

**Comment by Mardaneus86** [comment]
Looks good now, if you can just remove the `code.js` file I can accept the PR. You can use a folder named 'local' to save these sort of files. Contents of this folder are ignored by git.

**Comment by Kyderman** [comment]
Funnily enough added the git ignore after my commit, oops. Ill sort it now

**Comment by Kyderman** [comment]
Completed, will look at improvements in the future:

- Club value on top bar
- button press to initiate club value calc
- refresh club value button (after initial press)

**Pull request merged**